gceremoveimg

```diff
@@ -0,0 +1,157 @@
+argparse.add_argument(
+    '--all',
+    action='store_true',
+    default=False,
+    dest='all',
+    help='Delete all images that match the search criteria'
+)
+argparse.add_argument(
+    '--confirm',
+    action='store_true',
+    default=False,
+    dest='confirm',
+    help='Remove matched images with confirmation of action'
+)
+argparse.add_argument(
+    '-c', '--credentials-file',
+    dest='credentials_file_path',
+    help='Path to the service account credentials file',
+    metavar='CREDENTIALS_FILE'
+)
+help_msg = 'Do not perform any action, print information about actions that '
+help_msg += 'would be performed instead (Optional)'
+argparse.add_argument(
+    '-n', '--dry-run',
+    action='store_true',
+    default=False,
+    dest='dry_run',
+    help=help_msg
+)
+argparse.add_argument(
+    '-p', '--project',
+    dest='project_name',
+    help='Project to use for image deletion',
+    metavar='PROJECT_NAME',
+    required=True
+)
+# Note, one of the arguments in the group is required if --version is
+# not specified. However setting this behavior through the parser
+# also requires one argument to be specified even if --version is specified
+# This parser behavior is true even if --version and the group are part of the
+# same subgroup
+remove_image_condition_group = argparse.add_mutually_exclusive_group()
+remove_image_condition_group.add_argument(
+    '--image-name',
+    dest='image_name',
+    help='The image name of the image to be removed (Optional)',
+    metavar='IMAGE_NAME'
+)
+help_msg = 'An image name fragment to match the image name of the image to be '
+help_msg += 'removed (Optional)'
+remove_image_condition_group.add_argument(
+    '--image-name-frag',
+    dest='image_name_frag',
+    help=help_msg,
+    metavar='IMAGE_NAME_FRAGMENT'
+)
+help_msg = 'A regular expression to match the image name of the image to be '
+help_msg += 'removed (Optional)'
+remove_image_condition_group.add_argument(
+    '--image-name-match',
+    dest='image_name_match',
+    help=help_msg,
+    metavar='REGEX'
+)
+argparse.add_argument(
+    '--verbose',
+    action='store_true',
+    default=False,
+    dest='verbose',
+    help='Enable on verbose output'
+)
+argparse.add_argument(
+    '--version',
+    action='version',
+    version=utils.get_version(),
+    help='Program version'
+)
+
+args = argparse.parse_args()
+
+logger = utils.get_logger(args.verbose)
+
+# Explicit check required to to the group issue, see comment above
+if (
+        not args.image_name and not
+        args.image_name_frag and not
+        args.image_name_match):
+    error_msg = 'gceremoveimg: error: one of the arguments '
+    error_msg += '--image-name --image-name-frag '
+    error_msg += '--image-name-match is required'
+    logger.error(error_msg)
+    sys.exit(1)
+
+credentials_file = None
+if args.credentials_file_path:
+    credentials_file = os.path.expanduser(args.credentials_file_path)
```

Comment:
Ideally any helper functionality would happen in the class so that it doesn't need duplication when used as an API.